### PR TITLE
Specify locale for case conversion

### PR DESCRIPTION
**Pull request by elharo**
https://stackoverflow.com/questions/11063102/using-locales-with-javas-tolowercase-and-touppercase

### CODE REVIEW

**Comment by slachiewicz** [comment]
@michael-o English or like we have in Doxia ROOT ?

File: src/test/java/org/apache/maven/plugins/javadoc/JavadocReportTest.java [hunk 1]
```diff
@@ -427,31 +428,31 @@ public void testCustomConfiguration() throws Exception {
...
         // footer
         if (JavaVersion.JAVA_VERSION.isBefore("16-ea")
                 && !System.getProperty("java.vm.name").contains("OpenJ9")) {
-            assertTrue(str.toUpperCase().contains("MAVEN JAVADOC PLUGIN TEST FOOTER"));
+            assertTrue(str.toUpperCase(Locale.ENGLISH).contains("MAVEN JAVADOC PLUGIN TEST FOOTER"));
         }
 
         // nohelp == true
-        assertFalse(str.toUpperCase().contains("/HELP-DOC.HTML"));
+        assertFalse(str.toUpperCase(Locale.ENGLISH).contains("/HELP-DOC.HTML"));
```
Comment (michael-o) [inline src/test/java/org/apache/maven/plugins/javadoc/JavadocReportTest.java:455]: Is this really uppercase, that looks weird...

**Comment by michael-o** [comment]
> @michael-o English or like we have in Doxia ROOT ?

This is unrelated to Doxia. Here both English and `ROOT` are fine. I'd simply check other plugins for packaging comparison and would do it accordingly.

File: src/test/java/org/apache/maven/plugins/javadoc/JavadocReportTest.java (hunk 1 above)
Comment (elharo) [reply src/test/java/org/apache/maven/plugins/javadoc/JavadocReportTest.java:455]: Could go either way. 

assertFalse(str.toLowerCase(Locale.ENGLISH).contains("/help-doc.html"));

is semantically equivalent. Probably just the preference of whoever first wrote this line of code.

**Review by slachiewicz — APPROVED** [review]

**Pull request merged**